app/src/main/java/io/homeassistant/companion/android/sensors/StorageSensorManager.kt

```diff
@@ -59,12 +76,12 @@ class StorageSensorManager : SensorManager {
         var totalInternalStorage = getTotalInternalMemorySize()
         var freeInternalStorage = getAvailableInternalMemorySize()
         val percentageFreeInternalStorage = getPercentageInternal()
-        val hasExternalStorage = externalMemoryAvailable()
+        externalPath = externalMemoryAvailable(context)
         var totalExternalStorage = "No SD Card"
         var freeExternalStorage = "No SD Card"
 
-        if (hasExternalStorage) {
-            val pathSD = context.getExternalFilesDir(null)
+        if (externalPath != null) {
+            val pathSD = context.getExternalFilesDir(externalPath.toString())
             val statSD = StatFs(pathSD?.path)
             blockSizeSD = statSD.blockSizeLong

```

Comment:
While debugging:
`val pathSD = context.getExternalFilesDir(externalPath.toString())` becomes `/storage/emulated/0/Android/data/io.homeassistant.companion.android.debug/files/storage/8BF3-12EB/Android/data/io.homeassistant.companion.android.debug/files`

I was able to get the correct numbers with this: 

```suggestion
            val statSD = StatFs(externalPath.toString())
            blockSizeSD = statSD.blockSizeLong
```